CMakeLists.txt

```diff
@@ -10,7 +10,7 @@
 ###############################################################################
 # Minimum CMake version and policies
 ###############################################################################
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
```

Comment:
We can't upgrade TravisCI to use a newer distro unfortunately.

I run Arch Linux so everything is bleeding edge :)

However I don't want to break the build on Ubuntu 14.04LTS for no good reason. The change you are trying to make here can easily be done without requiring a new CMake version.

If there was a "killer feature" in a newer CMake version that it turns out we really need I would consider upgrading the minimum CMake version.